TiCDC create changefeed successfully but report error log

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
 Execute this cli: /cdc cli changefeed create --pd=http://lx-40-nightly-pd:2379 --sink-uri="mysql://root:mysql@172.16.4.29:30306"  --changefeed-id
="simple-replication-task"
[2021/02/04 09:51:08.816 +00:00] [ERROR] [mysql.go:139] ["flush row changed events routine starts failed"] [error="[CDC:ErrOperateOnClosedNotifier]operate on a closed notifier"] [errorVerbose="[CDC:ErrOperateOnClosedNotifier]operate on a closed notifier\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/ticdc/pkg/notify.(*Notifier).NewReceiver\n\tgithub.com/pingcap/ticdc/pkg/notify/notify.go:87\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).flushRowChangedEvents\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:137\nruntime.goexit\n\truntime/asm_amd64.s:1373"]
Create changefeed successfully!
ID: simple-replication-task
Info: {"sink-uri":"mysql://root:mysql@172.16.4.29:30306","opts":{"_changefeed_id":"cli-verify"},"create-time":"2021-02-04T09:51:08.787441081Z","start-ts":422689444663132162,"target-ts":0,"admin-job-type":0,"sort-engine":"memory","sort-dir":".","config":{"case-sensitive":true,"enable-old-value":false,"force-replicate":false,"check-gc-safe-point":true,"filter":{"rules":["*.*"],"ignore-txn-start-ts":null,"ddl-allow-list":null},"mounter":{"worker-num":16},"sink":{"dispatchers":null,"protocol":"default"},"cyclic-replication":{"enable":false,"replica-id":0,"filter-replica-ids":null,"id-buckets":0,"sync-ddl":false},"scheduler":{"type":"table-number","polling-time":-1}},"state":"normal","history":null,"error":null,"sync-point-enabled":false,"sync-point-interval":600000000000}
/ # /cdc cli changefeed list --pd=http://lx-40-nightly-pd:2379
[
  {
    "id": "simple-replication-task",
    "summary": {
      "state": "normal",
      "tso": 422689481979068417,
      "checkpoint": "2021-02-04 09:53:31.114",
      "error": null
    }
  }
]


2. What did you expect to see?
No error log since it create successfully.

3. What did you see instead?
[2021/02/04 09:51:08.816 +00:00] [ERROR] [mysql.go:139] ["flush row changed events routine starts failed"] [error="[CDC:ErrOperateOnClosedNotifier]operate on a closed notifier"] [errorVerbose="[CDC:ErrOperateOnClosedNotifier]operate on a closed notifier\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20201126102027-b0a155152ca3/normalize.go:156\ngithub.com/pingcap/ticdc/pkg/notify.(*Notifier).NewReceiver\n\tgithub.com/pingcap/ticdc/pkg/notify/notify.go:87\ngithub.com/pingcap/ticdc/cdc/sink.(*mysqlSink).flushRowChangedEvents\n\tgithub.com/pingcap/ticdc/cdc/sink/mysql.go:137\nruntime.goexit\n\truntime/asm_amd64.s:1373"]
Create changefeed successfully!



4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
     Release Version: v4.0.10
      Edition: Community
     Git Commit Hash: dbade8cda4c5a329037746e171449e0a1dfdb8b3
     Git Branch: heads/refs/tags/v4.0.10
     UTC Build Time: 2021-01-14 12:31:23
     GoVersion: go1.13
     Race Enabled: false
     TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
     Check Table Before Drop: false |


        ```

    - TiCDC version (execute `cdc version`):

        ```
        Release Version: v4.0.10
        Git Commit Hash: bced60a8d65f31ab68a35f1f309ea2345076096c
        Git Branch: HEAD
        UTC Build Time: 2021-01-15 05:11:33  
       Go Version: go version go1.14.13 linux/amd64

        ```

--- a/cdc/sink/mysql.go
+++ b/cdc/sink/mysql.go
@@ -132,12 +132,7 @@ func (s *mysqlSink) FlushRowChangedEvents(ctx context.Context, resolvedTs uint64
 	return checkpointTs, nil
 }
 
-func (s *mysqlSink) flushRowChangedEvents(ctx context.Context) {
-	receiver, err := s.resolvedNotifier.NewReceiver(50 * time.Millisecond)
-	if err != nil {
-		log.Error("flush row changed events routine starts failed", zap.Error(err))
-		return
-	}
+func (s *mysqlSink) flushRowChangedEvents(ctx context.Context, receiver *notify.Receiver) {
 	for {
 		select {
 		case <-ctx.Done():
@@ -616,7 +611,11 @@ func newMySQLSink(
 		return nil, err
 	}
 
-	go sink.flushRowChangedEvents(ctx)
+	receiver, err := sink.resolvedNotifier.NewReceiver(50 * time.Millisecond)
+	if err != nil {
+		return nil, err
+	}
+	go sink.flushRowChangedEvents(ctx, receiver)
 
 	return sink, nil
 }

--- a/cdc/sink/mysql_test.go
+++ b/cdc/sink/mysql_test.go
@@ -1211,3 +1211,43 @@ func (s MySQLSinkSuite) TestNewMySQLSinkExecDDL(c *check.C) {
 	err = sink.Close()
 	c.Assert(err, check.IsNil)
 }
+
+func (s MySQLSinkSuite) TestNewMySQLSink(c *check.C) {
+	defer testleak.AfterTest(c)()
+
+	dbIndex := 0
+	mockGetDBConn := func(ctx context.Context, dsnStr string) (*sql.DB, error) {
+		defer func() {
+			dbIndex++
+		}()
+		if dbIndex == 0 {
+			// test db
+			db, err := mockTestDB()
+			c.Assert(err, check.IsNil)
+			return db, nil
+		}
+		// normal db
+		db, mock, err := sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual))
+		mock.ExpectClose()
+		c.Assert(err, check.IsNil)
+		return db, nil
+	}
+	backupGetDBConn := getDBConnImpl
+	getDBConnImpl = mockGetDBConn
+	defer func() {
+		getDBConnImpl = backupGetDBConn
+	}()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	changefeed := "test-changefeed"
+	sinkURI, err := url.Parse("mysql://127.0.0.1:4000/?time-zone=UTC&worker-count=4")
+	c.Assert(err, check.IsNil)
+	rc := config.GetDefaultReplicaConfig()
+	f, err := filter.NewFilter(rc)
+	c.Assert(err, check.IsNil)
+	sink, err := newMySQLSink(ctx, changefeed, sinkURI, f, rc, map[string]string{})
+	c.Assert(err, check.IsNil)
+	err = sink.Close()
+	c.Assert(err, check.IsNil)
+}

